src/main/java/com/commercetools/sync/commons/helpers/BaseReferenceResolver.java

```diff
@@ -18,9 +18,10 @@
  *            specified by the user, on reference resolution.
  */
 public abstract class BaseReferenceResolver<T, S extends BaseSyncOptions> {
+    public static final String BLANK_KEY_VALUE_ON_RESOURCE_IDENTIFIER = "The value of the 'key' field of the Resource"
```

Comment:
The message could be improved. Because there is no key field on the Reference. 
`The value of the 'key' field of the ResourceIdentifier is blank (null/empty).`

Reply:
184e2e5f

Reply:
`ResourceIdentifier` is a type, so I would use the type name instead of writing as `resource identifier`

Reply:
ea7bb7e4